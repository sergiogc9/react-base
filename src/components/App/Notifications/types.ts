import { Notification, QueueNotification } from '@src/types/notification';

export type StateProps = {
	notifications: QueueNotification[]
};

export type DispatchProps = {
	onAddNotification: (notification: Notification) => void,
	onUnqueueNotification: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps;
