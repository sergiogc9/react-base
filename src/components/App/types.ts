import { TenantObject } from "@src/class/Tenant";
import { UserObject } from "@src/class/User";
import { FacebookAuthStatus } from "@src/types/facebook";
import { TProps } from '@src/lib/i18n';

export type StateProps = {
	authenticated: boolean
	tenant: TenantObject | null
	user: UserObject | null
};

export type DispatchProps = {
	onAuth: () => void
};

export type OwnProps = {};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
