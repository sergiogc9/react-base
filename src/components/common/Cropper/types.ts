type Props = {
	/**
	 * The image in Base64. If null, no image has been set before.
	 */
	readonly image: string;
	/**
	 * A callback method called with the cropped image data
	 */
	readonly onImageCropped: (image: string) => void;
};

export type CropperProps = Props;
