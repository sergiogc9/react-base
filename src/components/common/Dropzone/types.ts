import { TProps } from '@src/lib/i18n';

export type AddFileError = {
	error: 'file_too_large' | 'incorrect_file_type'
	file: File
};

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	files: File[]
	multiple: boolean
	accept?: string
	maxSize?: number
	showError?: boolean
	onFilesAdded(files: File[]): void
	onAddFilesError(errors: AddFileError[]): void
	onRemoveFile(file: File): void
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
