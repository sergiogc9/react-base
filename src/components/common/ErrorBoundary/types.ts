export type ErrorBoundaryProps = React.PropsWithChildren;
