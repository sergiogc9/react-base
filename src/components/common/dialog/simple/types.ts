import { TProps } from '@src/lib/i18n';

export type StateProps = {};

export type DispatchProps = {};

export type OwnProps = {
	id: string
	text: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
