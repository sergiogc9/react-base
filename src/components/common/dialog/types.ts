import { TProps } from '@src/lib/i18n';

export type StateProps = {
};

export type DispatchProps = {
};

export type OwnProps = {
	id: string
	icon?: JSX.Element
	title: string
	content: JSX.Element
	onAccept?: () => void
	onAcceptText?: string
	onCancel?: () => void
	onCancelText?: string
};

export type ComponentProps = StateProps & DispatchProps & OwnProps & TProps;
