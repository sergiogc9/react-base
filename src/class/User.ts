export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
