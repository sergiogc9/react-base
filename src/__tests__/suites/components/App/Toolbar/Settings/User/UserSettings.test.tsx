import { mount } from "enzyme";
import React from "react";
import CircularProgress from "react-md/lib/Progress/CircularProgress";

import { UserObject } from "@src/class/User";
import UserSettings from "@src/components/App/Toolbar/Settings/User/UserSettings";
import { FacebookAccount } from "@src/types/facebook";
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const user = TestHelper.getUser();

const facebookAccounts: FacebookAccount[] = [
	{
		expires_at: new Date(1549287550563),
		id: "facebook-id",
		insights_user_id: "insights-user-id",
		instagramAccounts: [{
			id: "instagram-id",
			facebook_page_id: "facebook-page-id",
			facebook_page_name: "facebook-page-name",
			name: "Instagram name",
			profile_picture_url: "http://fakeurl.com",
			screen_name: "insta-user"
		}],
		is_valid: true,
		name: "facebook name",
		picture_url: "http://fakeurl.com",
		scope: "",
		updated_at: new Date(1549287550563)
	}
];

const timezones = ["Europe/Girona", "Europe/Palencia"];

let updateUserSettingsMock: jest.Mock;
let onCloseMock: jest.Mock;
let onFetchUserFacebookAccounts: jest.Mock;
let onDeauthorizeFacebookAccountMock: jest.Mock;
let onDeauthorizeInstagramAccountMock: jest.Mock;
let onFetchTimezones: jest.Mock;

describe("UserSettings", () => {
	let wrapper: any;

	beforeEach(() => {
		onCloseMock = jest.fn();
		onFetchUserFacebookAccounts = jest.fn();
		updateUserSettingsMock = jest.fn();
		onDeauthorizeFacebookAccountMock = jest.fn();
		onDeauthorizeInstagramAccountMock = jest.fn();
		onFetchTimezones = jest.fn();

		wrapper = mount(
			<UserSettings
				user={user}
				facebookAccounts={facebookAccounts}
				loading={false}
				onUpdateUserSettings={updateUserSettingsMock}
				onFetchUserFacebookAccounts={onFetchUserFacebookAccounts}
				onDeauthorizeFacebookAccount={onDeauthorizeFacebookAccountMock}
				onDeauthorizeInstagramAccount={onDeauthorizeInstagramAccountMock}
				onClose={onCloseMock}
				timezones={timezones}
				onFetchTimezones={onFetchTimezones}
			/>
		);
	});

	it("Test UserSettings component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test UserSettings component without facebookAccounts", () => {
		expect(onFetchUserFacebookAccounts).toHaveBeenCalledTimes(0);
		wrapper = mount(
			<UserSettings
				user={user}
				facebookAccounts={null}
				loading={false}
				onUpdateUserSettings={updateUserSettingsMock}
				onFetchUserFacebookAccounts={onFetchUserFacebookAccounts}
				onDeauthorizeFacebookAccount={onDeauthorizeFacebookAccountMock}
				onDeauthorizeInstagramAccount={onDeauthorizeInstagramAccountMock}
				onClose={onCloseMock}
				timezones={timezones}
				onFetchTimezones={onFetchTimezones}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
		expect(onFetchUserFacebookAccounts).toHaveBeenCalledTimes(1);
	});

	it("Test UserSettings component without timezones", () => {
		expect(onFetchTimezones).toHaveBeenCalledTimes(0);
		wrapper = mount(
			<UserSettings
				user={user}
				facebookAccounts={facebookAccounts}
				loading={false}
				onUpdateUserSettings={updateUserSettingsMock}
				onFetchUserFacebookAccounts={onFetchUserFacebookAccounts}
				onDeauthorizeFacebookAccount={onDeauthorizeFacebookAccountMock}
				onDeauthorizeInstagramAccount={onDeauthorizeInstagramAccountMock}
				onClose={onCloseMock}
				timezones={null}
				onFetchTimezones={onFetchTimezones}
			/>
		);
		expect(wrapper.html()).toMatchSnapshot();
		expect(onFetchTimezones).toHaveBeenCalledTimes(1);
	});

	it("Test save user settings button handler when no changes", () => {
		wrapper.find('Button#settingsUserSaveButton').simulate('click');
		expect(updateUserSettingsMock).toHaveBeenCalledTimes(0);
	});

	it("Test save user settings button handler when changes", () => {
		wrapper.setProps({
			user: {
				settings: {
					language_code: 'es'
				}
			}
		});
		wrapper.find('Button#settingsUserSaveButton').simulate('click');
		expect(updateUserSettingsMock).toHaveBeenCalledTimes(1);
	});

	it("Test cancel button handler", () => {
		wrapper.find('Button#settingsUserCancelButton').simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
	});

	it("Test close (X) button handler", () => {
		wrapper.find('Button#userSettingsCloseButton').simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
	});

	it("Test facebookAccounts is loading", () => {
		wrapper.setProps({
			loading: true
		});
		expect(wrapper.find(CircularProgress)).toHaveLength(1);
		expect(wrapper.find('.settings-section')).toHaveLength(0);
	});

	it('Test new tenant index permissions', () => {
		const userWithoutPermissions: UserObject = {
			...user,
			permissions: []
		};

		// Mount again because user prop is assigned to state in constructor
		wrapper = mount(
			<UserSettings
				user={userWithoutPermissions}
				facebookAccounts={facebookAccounts}
				loading={false}
				onUpdateUserSettings={updateUserSettingsMock}
				onFetchUserFacebookAccounts={onFetchUserFacebookAccounts}
				onDeauthorizeFacebookAccount={onDeauthorizeFacebookAccountMock}
				onDeauthorizeInstagramAccount={onDeauthorizeInstagramAccountMock}
				onClose={onCloseMock}
				timezones={timezones}
				onFetchTimezones={onFetchTimezones}
			/>
		);
		expect(wrapper.find('SelectionControl#settingsUserSelectionControl')).toHaveLength(0);
	});

	it('Test change language input', () => {
		expect(wrapper.state().user.settings.language_code).toStrictEqual('en');
		wrapper.find("#settingsUserLanguageSelect").at(0).props().onChange("es");
		expect(wrapper.state().user.settings.language_code).toStrictEqual('es');
	});

	it('Test change locale', () => {
		expect(wrapper.state().user.settings.locale).toStrictEqual('en-GB');
		wrapper.find("#settingsUserLocaleSelect").at(0).props().onChange("es-ES");
		expect(wrapper.state().user.settings.locale).toStrictEqual('es-ES');
	});

	it('Test change time zone', () => {
		expect(wrapper.state().user.settings.timezone).toStrictEqual('UTC');
		wrapper.find("#settingsUserTimezoneSelect").at(0).props().onChange("Europe/Madrid");
		expect(wrapper.state().user.settings.timezone).toStrictEqual('Europe/Madrid');
	});

	it("Test click deauthorize facebook event handler", () => {
		wrapper.find('.facebook-account button').simulate('click');
		expect(onDeauthorizeFacebookAccountMock).toHaveBeenCalled();
	});

	it("Test click deauthorize instagram event handler", () => {
		wrapper.find('.instagram-account button').simulate('click');
		wrapper.find('button#launchmetricsDialogAcceptBtn').simulate('click'); // Dialog accept
		expect(onDeauthorizeInstagramAccountMock).toHaveBeenCalled();
	});

	it("Test deauthorize instagram dialog opened but cancelled", () => {
		wrapper.find('.instagram-account button').simulate('click');
		wrapper.find('button#launchmetricsDialogCancelBtn').simulate('click'); // Dialog accept
		expect(wrapper.html()).toMatchSnapshot();
	});
});
