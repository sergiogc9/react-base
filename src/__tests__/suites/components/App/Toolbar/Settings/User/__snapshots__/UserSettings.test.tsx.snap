// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettings Test UserSettings component snapshot 1`] = `"<div id=\\"settingsUser\\"><div class=\\"header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>account.value</span><button id=\\"userSettingsCloseButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.settings</div><div class=\\"section-content md-grid md-grid--stacked\\"><div id=\\"settingsUserLanguageSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLanguageSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.language.value</label><div id=\\"settingsUserLanguageSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.language.en</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLanguageSelect\\" value=\\"en\\"></div></div></div><div id=\\"settingsUserLocaleSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLocaleSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.locale.value</label><div id=\\"settingsUserLocaleSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.locale.en-GB</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLocaleSelect\\" value=\\"en-GB\\"></div></div></div><div id=\\"settingsUserTimezoneSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserTimezoneSelect\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">account.timezone</label><div id=\\"settingsUserTimezoneSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserTimezoneSelect\\" value=\\"UTC\\"></div></div></div><div class=\\"buttons-container md-grid\\"><button id=\\"settingsUserCancelButton\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--red-ink\\"><div class=\\"md-ink-container\\"></div>filters.period.custom.dialog.cancel</button><button id=\\"settingsUserSaveButton\\" type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block btn--green-ink\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">filters.period.custom.dialog.ok</span></div></button></div></div></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.facebook_accounts.title</div><div class=\\"section-content md-grid md-grid--stacked\\"><div><div class=\\"facebook-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">facebook name</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>account.instagram_accounts.unlink</button></div><div><div class=\\"connected-instagrams-text\\">account.instagram_accounts.title</div><div class=\\"instagram-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">Instagram name - insta-user</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div> account.instagram_accounts.unlink</button></div></div></div><div class=\\"add-facebook-account\\"><a href=\\"https://localhost:5000/facebook/connect?fromUrl=http%3A%2F%2Flocalhost%2F\\"><button id=\\"settingsUserAddFacebookButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">add</i></button></a></div></div></div></div>"`;

exports[`UserSettings Test UserSettings component without facebookAccounts 1`] = `"<div id=\\"settingsUser\\"><div class=\\"header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>account.value</span><button id=\\"userSettingsCloseButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><svg id=\\"settingsUserProgress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"md-progress md-progress--circular md-block-centered md-progress--circular-indeterminate\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 66 66\\"><circle class=\\"md-circular-progress-path md-circular-progress-path--animated\\" stroke-width=\\"6\\" stroke-linecap=\\"round\\" cx=\\"33\\" cy=\\"33\\" r=\\"30\\"></circle></svg></div>"`;

exports[`UserSettings Test UserSettings component without timezones 1`] = `"<div id=\\"settingsUser\\"><div class=\\"header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>account.value</span><button id=\\"userSettingsCloseButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.settings</div><div class=\\"section-content md-grid md-grid--stacked\\"><div id=\\"settingsUserLanguageSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLanguageSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.language.value</label><div id=\\"settingsUserLanguageSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.language.en</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLanguageSelect\\" value=\\"en\\"></div></div></div><div id=\\"settingsUserLocaleSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLocaleSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.locale.value</label><div id=\\"settingsUserLocaleSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.locale.en-GB</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLocaleSelect\\" value=\\"en-GB\\"></div></div></div><div id=\\"settingsUserTimezoneSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserTimezoneSelect\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">account.timezone</label><div id=\\"settingsUserTimezoneSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserTimezoneSelect\\" value=\\"UTC\\"></div></div></div><div class=\\"buttons-container md-grid\\"><button id=\\"settingsUserCancelButton\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--red-ink\\"><div class=\\"md-ink-container\\"></div>filters.period.custom.dialog.cancel</button><button id=\\"settingsUserSaveButton\\" type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block btn--green-ink\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">filters.period.custom.dialog.ok</span></div></button></div></div></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.facebook_accounts.title</div><div class=\\"section-content md-grid md-grid--stacked\\"><div><div class=\\"facebook-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">facebook name</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>account.instagram_accounts.unlink</button></div><div><div class=\\"connected-instagrams-text\\">account.instagram_accounts.title</div><div class=\\"instagram-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">Instagram name - insta-user</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div> account.instagram_accounts.unlink</button></div></div></div><div class=\\"add-facebook-account\\"><a href=\\"https://localhost:5000/facebook/connect?fromUrl=http%3A%2F%2Flocalhost%2F\\"><button id=\\"settingsUserAddFacebookButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">add</i></button></a></div></div></div></div>"`;

exports[`UserSettings Test deauthorize instagram dialog opened but cancelled 1`] = `"<div id=\\"settingsUser\\"><div class=\\"header md-grid\\"><i class=\\"md-icon material-icons\\">person</i><span>account.value</span><button id=\\"userSettingsCloseButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-text--theme-primary md-ink--primary md-inline-block close-btn btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">clear</i></button></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.settings</div><div class=\\"section-content md-grid md-grid--stacked\\"><div id=\\"settingsUserLanguageSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLanguageSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.language.value</label><div id=\\"settingsUserLanguageSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.language.en</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLanguageSelect\\" value=\\"en\\"></div></div></div><div id=\\"settingsUserLocaleSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserLocaleSelect\\" class=\\"md-floating-label md-floating-label--floating md-text--secondary\\">account.locale.value</label><div id=\\"settingsUserLocaleSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field md-text\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\">account.locale.en-GB</span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserLocaleSelect\\" value=\\"en-GB\\"></div></div></div><div id=\\"settingsUserTimezoneSelect-menu\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu--select-field md-cell select\\"><div class=\\"md-select-field__toggle\\"><label for=\\"settingsUserTimezoneSelect\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">account.timezone</label><div id=\\"settingsUserTimezoneSelect-toggle\\" role=\\"listbox\\" tabindex=\\"0\\" aria-pressed=\\"false\\" class=\\"md-paper md-paper--0 md-fake-btn md-pointer--hover md-fake-btn--no-outline md-select-field\\"><div class=\\"md-icon-separator md-text-field md-text-field--floating-margin md-select-field--text-field\\"><span class=\\"md-icon-text\\"></span><i class=\\"md-icon material-icons\\">arrow_drop_down</i></div><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left md-divider--select-field\\"><input type=\\"hidden\\" id=\\"settingsUserTimezoneSelect\\" value=\\"UTC\\"></div></div></div><div class=\\"buttons-container md-grid\\"><button id=\\"settingsUserCancelButton\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--red-ink\\"><div class=\\"md-ink-container\\"></div>filters.period.custom.dialog.cancel</button><button id=\\"settingsUserSaveButton\\" type=\\"button\\" class=\\"md-btn md-btn--raised md-btn--text md-pointer--hover md-text md-inline-block btn--green-ink\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-icon-separator\\"><i class=\\"md-icon material-icons md-text--inherit\\">save</i><span class=\\"md-icon-text\\">filters.period.custom.dialog.ok</span></div></button></div></div></div><div class=\\"settings-section\\"><div class=\\"section-title\\">account.facebook_accounts.title</div><div class=\\"section-content md-grid md-grid--stacked\\"><div><div class=\\"facebook-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">facebook name</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div>account.instagram_accounts.unlink</button></div><div><div class=\\"connected-instagrams-text\\">account.instagram_accounts.title</div><div class=\\"instagram-account md-grid\\"><img src=\\"http://fakeurl.com\\" alt=\\"\\"><div class=\\"name\\">Instagram name - insta-user</div><button type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block\\"><div class=\\"md-ink-container\\"></div> account.instagram_accounts.unlink</button></div></div></div><div class=\\"add-facebook-account\\"><a href=\\"https://localhost:5000/facebook/connect?fromUrl=http%3A%2F%2Flocalhost%2F\\"><button id=\\"settingsUserAddFacebookButton\\" type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-inline-block\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">add</i></button></a></div></div></div></div>"`;
